metashare/api/models.py

```diff
@@ -184,6 +197,15 @@ class Meta:
         ordering = ("name",)
         unique_together = (("repo_owner", "repo_name"),)
 
+    def save(self, *args, **kwargs):
+        if not self.branch_name:
+            user = self.get_a_matching_user()
+            if user:
+                repo = gh.get_repo_info(user, repo_id=self.id)
+                self.branch_name = repo.default_branch
```

Comment:
There are a lot of other places we're using `repo.default_branch` directly. It seems a bit odd to me to have two sources for the same piece of information... but I'm guessing there's a good reason to do so?

Reply:
The motivation was so that we wouldn't have to make a GitHub API call in a model attribute to get the base for a project PR. I'm not sure it's a _good_ reason but it felt better than setting up a call to the GitHub API in that attribute.

Reply:
Right, makes sense. In that case would it be better for MS to consistently use `repo.branch_name` once we store it, or even occasionally check whether `repo.default_branch == repo.branch_name`?

Reply:
I assume those are two different values of `repo`: one is a `github3.Repository` and the other is a `metashare.api.models.Repository`.

Reply:
OK, it's already the case that everywhere where we have the `metashare.api.models.Repository` object handy, we use the `branch_name` from it, and when we only have a `github3.Repository`, we use the `default_branch` off of it.

Reply:
I'm okay with that. Do we want to ever check to see if `default_branch` has changed?

Reply:
I'm still pondering when that would make sense.